### ISSUE TITLE
[MapControl] Setting MapStyle property does not change MapStyle in .NET Core WPF App

### ISSUE THREAD

**Issue by luisquintanilla** [windows-toolkit/Microsoft.Toolkit.Win32#205]
## Describe the bug

Using the following documentation:

- https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/host-standard-control-with-xaml-islands
- https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/wpf-winforms/mapcontrol

I was able to create a .NET Core WPF application that utilized a `MapControl`. However, when I try to set the `MapStyle` property to `Aerial` both in XAML and in code-behind, nothing actually happens. Changing coordinates and other functionality seems to work. However, it doesn't seem I am able to change the style. Not sure if I'm doing something wrong or if it's a bug. I have included a screenshot and the repo where the code is below.

- [ ] Is this bug a regression in the toolkit?  If so, what toolkit version did you last see it work:

## Screenshots

![image](https://user-images.githubusercontent.com/46974588/68988964-d33feb00-080d-11ea-90a3-3ec64bce4dc1.png)

## Environment

**WPF App NuGet Package(s)**:

```xml
<PackageReference Include="Microsoft.Extensions.Configuration" Version="3.0.0" />
<PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="3.0.0" /><PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="3.0.0" /> 
<PackageReference Include="Microsoft.Toolkit.Wpf.UI.Controls" Version="6.0.0" />
```

**UWP App NuGet Package(s)**:

```xml
<PackageReference Include="Microsoft.Toolkit.Win32.UI.XamlApplication">
  <Version>6.0.0</Version>
</PackageReference>
```
Project .NET Version:
- [ ] .NET Framework (version: )
- [x] .NET Core 3
- [ ] .NET Core 3.1 Preview (version: )

Windows 10 Build Number:
- [ ] Fall Creators Update (16299)
- [ ] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [x] May 2019 Update (18362)
- [ ] Insider Build (build number: )

App min and target version:
- [ ] Fall Creators Update (16299)
- [ ] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [x] May 2019 Update (18362)
- [ ] Insider Build (xxxxx)

Device form factor:
- [x] Desktop
- [ ] Xbox
- [ ] Surface Hub
- [ ] IoT

Visual Studio 
- [ ] 2017 (version: )
- [x] 2019 (version: 16.3.9) 
- [ ] 2019 Preview (version: )

## Additional context

**Repository**: https://github.com/luisquintanilla/WPFMappingApp3

**WPF App Source Code**: https://github.com/luisquintanilla/WPFMappingApp3/tree/master/WPFMappingApp3

`MapControl` is in *MainWindows.xaml*.

**UWP App Source Code**: https://github.com/luisquintanilla/WPFMappingApp3/tree/master/WPFMappingApp3.UWP

**Comment by luisquintanilla** [comment]
Updated version of the app. Issue still persists.

https://github.com/luisquintanilla/MapControlCoreWPFSample

**Comment by ocalvo** [comment]
@luisquintanilla Could try this:
var mapControl = ...
(mapControl.Child as Windows.UI.Xaml.Maps.MapControl).MapStyle = Windows.UI.Xaml.Maps.MapStyle.Aerial

**Comment by luisquintanilla** [comment]
@ocalvo where would this go. MainPage constructor?

**Comment by michael-hawker** [comment]
@luisquintanilla try putting it at the start of your loaded event: https://github.com/luisquintanilla/MapControlCoreWPFSample/blob/master/MapControlCoreWPFSample/MainWindow.xaml.cs#L18

You already have a reference to `MapControl` as you named it in XAML so it'd just be:

`(MapControl.Child as Windows.UI.Xaml.Maps.MapControl).MapStyle = Windows.UI.Xaml.Maps.MapStyle.Aerial`

The reason this isn't working is because the controls here are just dummy wrappers. Not all of them may be setup properly. If you look at the [MapControl class](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/blob/master/Microsoft.Toolkit.Wpf.UI.Controls/MapControl/MapControl.cs#L46) for instance, `MapStyle` isn’t bound on the Initialized method (It actually looks like it’s bound to ‘Style’ maybe instead?). That could indicate part of the issue? Or there could be an issue with how the property introps between the two layers. We’ve seen that in the past with the [MediaPlayer control](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/196/files).

If you have time to tinker with this and find a solution for the correct mapping for the control in the toolkit as well, please feel free to submit a PR. Our next release of the toolkit is at the end of May, but we’ll also be shipping previews of 7.0 if it requires a breaking change to fix.

**Comment by luisquintanilla** [comment]
Okay. Sounds good. Let me try that.

**Comment by luisquintanilla** [comment]
@ocalvo @michael-hawker I don't have access to the `Child` property. The only thing I have access to is `Children`. However, when I try that I can't cast it. Is there a package reference I need to be making? Right now, this is all I have in my csproj.
```xml
<ItemGroup>
    <PackageReference Include="Microsoft.Toolkit.Wpf.UI.Controls" Version="6.0.0" />
</ItemGroup>
```

**Comment by michael-hawker** [comment]
@luisquintanilla the `GetUwpInternalObject()` should be public, try that. I think all the other ones get masked by the parent implementations.

**Comment by marb2000** [comment]
@luisquintanilla did you solve/workaround the issue?

**Comment by luisquintanilla** [comment]
@marb2000 I have found a workaround. I am still having issues with my specific use case, but you can close this issue since now I'm trying to solve another problem. Thanks everyone for your help.

**Comment by asparrowhawk** [comment]
This issue doesn't appear to be fixed in version 6.1.2 of the _Microsoft.Toolkit.Wpf.UI.Controls_ NuGet package. Setting the Style property on the MapControl fails to update the underlying _Windows.UI.Xaml.Controls.Maps.MapControl_. This can be seen by inspecting the properties of the object returned from the call to _GetUwpInternalObject()_. It is only after casting the returned object and then updating the Style property that the MapControl rendering changes.
